src/mlpack/methods/decision_tree/gini_gain.hpp

```diff
@@ -44,37 +44,128 @@ class GiniGain
     if (labels.n_elem == 0)
       return 0.0;
 
-    // Count the number of elements in each class.
-    arma::vec counts(numClasses, arma::fill::zeros);
+    // Count the number of elements in each class.  Use four auxiliary vectors
```

Comment:
Really nice, maybe there are other places we can do the same, and get about the same speedup.